superset-frontend/src/views/CRUD/csstemplates/CssTemplateModal.tsx

```diff
@@ -0,0 +1,258 @@
+    fetchResource,
+    createResource,
+    updateResource,
+  } = useSingleViewResource<TemplateObject>(
+    'css_template',
+    t('css_template'),
+    addDangerToast,
+  );
+
+  // Functions
+  const hide = () => {
+    setIsHidden(true);
+    onHide();
+  };
+
+  const onSave = () => {
+    if (isEditMode) {
+      // Edit
+      if (currentCssTemplate && currentCssTemplate.id) {
+        const update_id = currentCssTemplate.id;
+        delete currentCssTemplate.id;
+        delete currentCssTemplate.created_by;
+        updateResource(update_id, currentCssTemplate).then(() => {
+          if (onCssTemplateAdd) {
+            onCssTemplateAdd();
+          }
+
+          hide();
+        });
+      }
+    } else if (currentCssTemplate) {
+      // Create
+      createResource(currentCssTemplate).then(() => {
+        if (onCssTemplateAdd) {
+          onCssTemplateAdd();
+        }
+
+        hide();
+      });
+    }
+  };
+
+  const onTemplateNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { target } = event;
+
+    const data = {
+      ...currentCssTemplate,
+      template_name: currentCssTemplate ? currentCssTemplate.template_name : '',
+      css: currentCssTemplate ? currentCssTemplate.css : '',
+    };
+
+    data[target.name] = target.value;
+    setCurrentCssTemplate(data);
+  };
+
+  const onCssChange = (css: string) => {
+    const data = {
+      ...currentCssTemplate,
+      template_name: currentCssTemplate ? currentCssTemplate.template_name : '',
+      css,
+    };
+    setCurrentCssTemplate(data);
+  };
+
+  const validate = () => {
+    if (
+      currentCssTemplate &&
+      currentCssTemplate.template_name.length &&
+      currentCssTemplate.css &&
+      currentCssTemplate.css.length
+    ) {
+      setDisableSave(false);
+    } else {
+      setDisableSave(true);
+    }
+  };
+
+  // Initialize
+  if (
+    isEditMode &&
+    (!currentCssTemplate ||
+      !currentCssTemplate.id ||
+      (cssTemplate && cssTemplate.id !== currentCssTemplate.id) ||
+      (isHidden && show))
+  ) {
+    if (cssTemplate && cssTemplate.id !== null && !loading) {
+      const id = cssTemplate.id || 0;
+
+      fetchResource(id).then(() => {
+        setCurrentCssTemplate(resource);
+      });
+    }
+  } else if (
+    !isEditMode &&
+    (!currentCssTemplate || currentCssTemplate.id || (isHidden && show))
+  ) {
+    setCurrentCssTemplate({
+      template_name: '',
+      css: '',
+    });
+  }
+
+  // Validation
+  useEffect(() => {
+    validate();
+  }, [
+    currentCssTemplate ? currentCssTemplate.template_name : '',
+    currentCssTemplate ? currentCssTemplate.css : '',
+  ]);
+
+  // Show/hide
+  if (isHidden && show) {
+    setIsHidden(false);
+  }
+
+  return (
+    <Modal
+      disablePrimaryButton={disableSave}
+      onHandledPrimaryAction={onSave}
+      onHide={hide}
+      primaryButtonName={isEditMode ? t('Save') : t('Add')}
+      show={show}
+      width="800px"
```

Comment:
FWIW, we can do some fun stuff with this, e.g. `width={"calc(100% - 100px)"}` if you ever feel like it :D

If/when there are a lot of these modals kicking around, we should probably add a limited number of preferred values to the Theme or a constants file.